### DISCUSSION TITLE
Unmatched Schmid's factor maps

### DISCUSSION THREAD

**Discussion by trudator** [mtex-toolbox/mtex#1706] [Ask Anything] [answered]
Hello,

I've been using MTEX for a while and just came across analyzing my data using the TSL software by EDAX. Though the settings (crystal symmetry, reference frame) were found to be similar, Schmid's factor maps seem to be considerably unmatched (see images below). My Schmid's factor was calculated upon the **compression uniaxial** deformation on the slip system of **{1 1 0}[-1 1 1]** for the BCC structure in both cases. I've tried to play around and around with those settings but ended up same results anytime. Any suggestions would be very much appreciated. Thanks.

My code in MTEX: 
```
%Initialize MTEX
startup_mtex;
setMTEXpref('xAxisDirection','south');
setMTEXpref('zAxisDirection','outOfPlane');
ebsd = EBSD.load(fname,CS,'interface','ang','convertSpatial2EulerReferenceFrame');
setMTEXpref('FontSize',14);   
Ini.dataPath = [pwd,'/data/'];
Ini.cifPath = [Ini.dataPath,'input/cif/'];
Ini.ebsdPath = [Ini.dataPath,'input/ebsd/'];
Ini.texturePath = [Ini.dataPath,'output/texture/'];

%Schmid factor calculation
sS = slipSystem(Miller(-1,1,1,ebsd('Martensite BCC').CS,'uvw'),Miller(1,1,0,ebsd('Martensite BCC').CS,'hkl'))
sS = sS.symmetrise;
sSLocal = grains('Martensite BCC').meanOrientation * sS
sigma = stressTensor.uniaxial(vector3d.Z);
SF = sSLocal.SchmidFactor(-sigma);
[SFMax,active] = max(abs(SF),[],2);
close all;
figure;
plot(grains('Martensite BCC'),SFMax,'micronbar','off','linewidth',2);
cm = [jet(280)];
mtexColorMap(cm);
mtexColorbar ('location','eastOutSide','title','Schmid factor','FontSize',15,'ticks',0:0.01:0.5);
```

**Comment by kilir** [comment, marked as answer]
Hi,
there are several things to consider:
a) are you sure you import your ebsd data correctly, i.e. correct coordinate system etc., such that the ebsd data is not just similar but identical. 
b) you are comparing a map of Schmid factors for each pixel against a grain map of Schmid factors. It might not be the easiest to compare.
c) I don't know anything about the specimen coordinate system in TSL, but I'd interpret the [1 0 0;0 0 0;0 0 0] as a uniaxial stress tensor pushing/pulling (depending your world) parallel to  the x-direction. In your code you use -zvector.
Cheers,
Rüdiger

**Reply by trudator** [reply]
Hi Rudiger,

I changed to -xvector as your suggestion and it's magically matched. I still don't get how you jumped to that assessment. In my TSL settings, I chose to set 2 as the specimen coordinate in which the Z-direction (A3) is pointing outside of the plane. So if a sample was uniaxially compressed, the tensor should be [0 0 0;0 0 0;0 0 -1]? 

![image](https://user-images.githubusercontent.com/117386810/211114320-83553094-a055-4a38-960e-ff7b5bd8e671.png)

**Reply by kilir** [reply]
The TSL stress tensor was written on your image.

I'd argue that in many cases  experimental, external, totally unrelated specimen directions may not match an SEM/EBSD coordinate system.  What you see there, is just explanation of how OIM matches coordinates in the specimen surface to the coordinates of the detector(orientations), those have (except you do in-situ experiments) nothing to do with how one does an experiment, cuts and mounts a sample.
Hope this helps.
Cheers,
Rüdiger

**Reply by trudator** [reply]
Thanks for the explanation. Do you think that I had set the wrong tensor direction in OIM TSL? My intention was to have the uniaxial compression load in the same direction as A3 (from setting 2). 

P/S: A bit of context, I can't access the TSL software again to look at what I configured. The huge difference in Schmid's factor map makes me confused if my settings in TSL OIM were wrong or not.

**Reply by kilir** [reply]
Maybe you'd have to consult the OIM documentation how the "sigma values" on the figure need top be interpreted.

If your orientation/ebsd data matches between OIM and mtex and you want to compute the Schmid factor for uniaxial compression along A3, that would be vector3d(0,0,1) in mtex.

**Reply by trudator** [reply]
Thanks Rudiger
